R/topics.R

```diff
@@ -167,3 +167,56 @@ topic_must <- function(message, topic) {
     x = paste0("Not ", encodeString(topic, quote = "'"))
   ))
 }
+
+content_info <- function(content_entry, index, pkg, section) {
+
+  if (grepl(".*::.*", content_entry)) {
+    names <- strsplit(content_entry, "::")[[1]]
+    if (!rlang::is_installed(names[1])) {
+      abort(
+        sprintf(
+          "%s must be installed if it is mentioned in the reference index.",
+          names[1]
+        )
+      )
+    }
+
+    path <- downlit::href_topic(names[2], names[1])
+    if (is.na(path)) {
+      abort(
+        sprintf(
+          "Could not find an href for topic %s of package %s",
+          names[2], names[1]
+        )
+      )
+    }
+
+    rd_name <- fs::path_ext_set(fs::path_file(path), "Rd")
+    file <- withr::local_tempfile(fileext = ".html")
+    write_lines(capture.output(tools::Rd2HTML(tools::Rd_db(names[1])[[rd_name]])), file)
+    rd <- xml2::read_html(file)
+
+
+    tibble::tibble(
+      path = path,
+      aliases = content_entry,
+      name = list(content_entry = NULL),
+      title = xml_text1(xml2::xml_find_first(rd, ".//h2")),
+      icon = list(content_entry = NULL)
+    )
+
+  } else {
```

Comment:
This is for the case that's the topic is not in a different package? That should be the primary case for this function (since it's the most important), so you should rewrite to make that more clear.